core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -810,19 +810,22 @@ class GroupMetadataManager(brokerId: Int,
    */
   private def maybeUpdateCoordinatorEpoch(
     partitionId: Int,
-    epochOpt: Option[Int]
+    epochOpt: OptionalInt
   ): Boolean = {
     val updatedEpoch = epochForPartitionId.compute(partitionId, (_, currentEpoch) => {
       if (currentEpoch == null) {
-        epochOpt.map(Int.box).orNull
+        if (epochOpt.isPresent) epochOpt.getAsInt
+        else null
       } else {
-        epochOpt match {
-          case Some(epoch) if epoch > currentEpoch => epoch
-          case _ => currentEpoch
-        }
+        if (epochOpt.isPresent && epochOpt.getAsInt > currentEpoch) epochOpt.getAsInt
+        else currentEpoch
       }
     })
-    epochOpt.forall(_ == updatedEpoch)
+    if(epochOpt.isPresent) {
```

Comment:
nit: spacing.

I also wonder if there is a cleaner way to write the above methods with the OptionalInt. But perhaps not.

Reply:
I see the options here are limited. Did you consider using java optionals here too? I suppose those are a bit more heavyweight and maybe still don't provide much better methods.

Reply:
> I also wonder if there is a cleaner way to write the above methods with the OptionalInt. But perhaps not.

This is the best I came up with.

> I see the options here are limited. Did you consider using java optionals here too? I suppose those are a bit more heavyweight and maybe still don't provide much better methods.

I don't understand your point here. We already use java optionals here now.

Reply:
There's minor method differences between OptionalInt and Optional<Integer>. That was my point. But maybe not big enough to change.

Reply:
Yeah. I was only suggesting because the other provides a few more methods (like getOrElse letting you return null instead of an int) but yeah, probably not worth changing.

Reply:
The point made by @jolshan looks helpful.